sonar-ballerina-plugin/src/main/resources/io/ballerina/sonar/ballerina-10.html

```diff
@@ -0,0 +1,13 @@
+<p>
+    Self-assignments, where a variable is assigned to itself (x = x), are redundant and do not alter the state of the variable.
+    They can indicate incomplete or erroneous logic and make the code harder to read and maintain.
```

Comment:
Shall we indicate that this may be a mistake done by the user

Reply:
`They can indicate incomplete or erroneous logic `, isn't this enough? Do we need phrase this better?